tokenclient/tokenFlows.go

```diff
@@ -31,6 +33,27 @@ type TokenFlows struct {
 	identity env.Identity
 	Options  Options
 	tokenURI string
+	cache    *cache.Cache
+}
+
+type request struct {
+	http.Request
+	key string `default:""`
+}
+
+func (r *request) cacheKey() string {
+	bodyReader, err := r.GetBody()
+	if err != nil {
+		panic("Unexpected error, can't read request body: " + err.Error())
+	}
+	params, err := io.ReadAll(bodyReader)
+	if err != nil {
+		panic("Unexpected error, can't read request body: " + err.Error())
+	}
+	if r.key == "" {
```

Comment:
I think it makes more sense to check `if r.key != ""` in the beginning of the func and return it then.
After you've recomputed the key this check is superfluous.

Reply:
absolutely.